src/block/Potato.php

```diff
@@ -23,16 +23,33 @@
 
 namespace pocketmine\block;
 
+use pocketmine\block\utils\FortuneTrait;
 use pocketmine\item\Item;
 use pocketmine\item\VanillaItems;
 use function mt_rand;
 
 class Potato extends Crops{
+	use FortuneTrait;
 
-	public function getDropsForCompatibleTool(Item $item) : array{
-		$result = [
-			VanillaItems::POTATO()->setCount($this->age >= self::MAX_AGE ? mt_rand(1, 5) : 1)
-		];
+	public function getPickedItem(bool $addUserData = false) : Item{
+		return VanillaItems::POTATO();
+	}
```

Comment:
This doesn't seem related to the PR. It's already handled by asItem().

Again, the diff for this file looks very chaotic due to logic moving around, making it difficult to read.